code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/ad/ADConfigManager.java

```diff
@@ -145,40 +163,77 @@ protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Obje
                 serviceAccountUsername, serviceAccountPassword, tls, userSearchField, userLoginField, userObjectClass,
                 userNameField, userEnabledAttribute, groupSearchField, groupObjectClass, groupNameField,
-                connectionTimeout);
+                connectionTimeout, identities);
     }
 
-    public ADConfig updateCurrentConfig(LDAPConstants config) {
-        settingsUtils.changeSetting(ADConstants.ACCESS_MODE_SETTING, config.getAccessMode());
-        settingsUtils.changeSetting(ADConstants.DOMAIN_SETTING, config.getDomain());
-        settingsUtils.changeSetting(ADConstants.GROUP_NAME_FIELD_SETTING, config.getGroupNameField());
-        settingsUtils.changeSetting(ADConstants.GROUP_OBJECT_CLASS_SETTING, config.getGroupObjectClass());
-        settingsUtils.changeSetting(ADConstants.GROUP_SEARCH_FIELD_SETTING, config.getGroupSearchField());
-        settingsUtils.changeSetting(ADConstants.LOGIN_DOMAIN_SETTING, config.getLoginDomain());
-        settingsUtils.changeSetting(ADConstants.PORT_SETTING, config.getPort());
-        settingsUtils.changeSetting(ADConstants.SERVER_SETTING, config.getServer());
-        settingsUtils.changeSetting(ADConstants.SERVICE_ACCOUNT_PASSWORD_SETTING, config.getServiceAccountPassword());
-        settingsUtils.changeSetting(ADConstants.SERVICE_ACCOUNT_USERNAME_SETTING, config.getServiceAccountUsername());
-        settingsUtils.changeSetting(ADConstants.TLS_SETTING, config.getTls());
-        settingsUtils.changeSetting(ADConstants.USER_DISABLED_BIT_MASK_SETTING, config.getUserDisabledBitMask());
-        settingsUtils.changeSetting(ADConstants.USER_ENABLED_ATTRIBUTE_SETTING, config.getUserEnabledAttribute());
-        settingsUtils.changeSetting(ADConstants.USER_LOGIN_FIELD_SETTING, config.getUserLoginField());
-        settingsUtils.changeSetting(ADConstants.USER_NAME_FIELD_SETTING, config.getUserNameField());
-        settingsUtils.changeSetting(ADConstants.USER_OBJECT_CLASS_SETTING, config.getUserObjectClass());
-        settingsUtils.changeSetting(ADConstants.USER_SEARCH_FIELD_SETTING, config.getUserSearchField());
-        settingsUtils.changeSetting(ADConstants.TIMEOUT_SETTING, config.getConnectionTimeout());
-        settingsUtils.changeSetting(SecurityConstants.SECURITY_SETTING, config.getEnabled());
-        if (config.getEnabled() != null){
+    public ADConfig updateCurrentConfig(Map<String, Object> config) {
+        settingsUtils.changeSetting(ADConstants.ACCESS_MODE_SETTING, config.get(AbstractTokenUtil.ACCESSMODE));
+        settingsUtils.changeSetting(ADConstants.DOMAIN_SETTING, config.get(DOMAIN));
+        settingsUtils.changeSetting(ADConstants.GROUP_NAME_FIELD_SETTING, config.get("groupNameField"));
+        settingsUtils.changeSetting(ADConstants.GROUP_OBJECT_CLASS_SETTING, config.get("groupObjectClass"));
+        settingsUtils.changeSetting(ADConstants.GROUP_SEARCH_FIELD_SETTING, config.get("groupSearchField"));
```

Comment:
I am seeing that each config field is being referenced in the hardcoded string twice (here and in ADConfigManager.java). Could we move it to the constants fields?